src/main/java/org/opensearch/performanceanalyzer/writer/EventLogQueueProcessor.java

```diff
@@ -172,8 +172,11 @@ private void writeAndRotate(
         if (lastTimeBucket != 0 && lastTimeBucket != currTimeBucket) {
             eventLogFileHandler.renameFromTmp(lastTimeBucket);
         }
-        // This appends the data to a file named <currTimeBucket>.tmp
-        eventLogFileHandler.writeTmpFile(currMetrics, currTimeBucket);
+        // Append to the tmp file only if we have metrics to publish.
```

Comment:
This check is too late into the code execution. 

We can have it inside `purgeQueueAndPersist()` on line 116 after checking  PA is disabled. We check `PerformanceAnalyzerMetrics.metricQueue.size() == 0` , we write a log message, emit metric and return.  Doing this would mean the `.tmp` file from previous window willl not be rotated and will remain on filesystem, which is accetable as the `MetricPurgeActivity` will eventually clean up the tmp files.

Reply:
Thought towards not adding this in the ```purgeQueueAndPersist``` method is that it prevents rotating the ```.tmp``` file from the previous window (metrics might or might not be present in this window). 

In this way, any absence of metrics in the current or the next window will not impact the previous window rotation. Any thoughts?

Reply:
Agreed.
The trade-off between 'returning early would mean we do not rotate the previous `.tmp` file and we lose one sampling window metrics' versus 'more readable and clean code'.
Would leave the decision to you.